### PR TITLE
feat(jest): mithril component testing

### PR DESCRIPTION
**Pull request by SychO9**
**Changes proposed in this pull request:**
This is part two of adding jest testing. It makes the necessary changes to be able to test component behavior.

**Reviewers should focus on:**
Part one is here: #3678

**Necessity**

- [ ] Has the problem that is being solved here been clearly explained?
- [ ] If applicable, have various options for solving this problem been considered?
- [ ] For core PRs, does this need to be in core, or could it be in an extension?
- [ ] Are we willing to maintain this for years / potentially forever?

**Confirmed**

- [ ] Frontend changes: tested on a local Flarum installation.
- [ ] Backend changes: tests are green (run `composer test`).
- [ ] Core developer confirmed locally this works as intended.
- [ ] Tests have been added, or are not appropriate here.

**Required changes:**

- [ ] Related documentation PR: (Remove if irrelevant)
- [ ] Related core extension PRs: (Remove if irrelevant)

### CODE REVIEW

File: js-packages/jest-config/test-matchers.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { expect } from '@jest/globals';
+
+// Extend Jest with mithril component test matchers.
+expect.extend({
+  toHaveElement: intoMatcher((out: any, expected: any) => out.should.have(expected), 'Expected $received to have node $expected'),
```
Comment (askvortsov1) [inline js-packages/jest-config/test-matchers.ts:5]: Can we add a test that uses these to confirm that they work?

**Review by askvortsov1 — APPROVED** [review]
This is awesome!!!

**Comment by askvortsov1** [comment]
Is testing supported in CI? I'm not seeing corresponding jobs in https://github.com/flarum/framework/actions/runs/3456520912/jobs/5769312494

**Comment by SychO9** [comment]
> Is testing supported in CI? I'm not seeing corresponding jobs in https://github.com/flarum/framework/actions/runs/3456520912/jobs/5769312494

We can look into it after it's merged (https://github.com/flarum/action-build/pull/12) I don't remember if I tested the CI changes or not.

**Pull request merged**